src/memory/client/MemoryBrowser.tsx

```diff
@@ -101,14 +124,28 @@ export class MemoryBrowser extends React.Component<Props, State> {
           error: undefined,
           memory: undefined,
         });
-        const result = await messageBroker.send({
-          command: 'ReadMemory',
-          args: {
-            address: this.addressReq,
-            length: parseInt(this.lengthReq),
-          },
-        });
-        this.setState({ memory: result.result });
+        if (this.state.childrenNames.length > 0) {
+          const result = await messageBroker.send({
+            command: 'ReadMemory',
+            args: {
+              address: this.addressReq,
+              length: parseInt(this.lengthReq),
+              child: this.state.childrenNames.length
+                ? this.childReq
+                : undefined,
```

Comment:
You almost collapsed the two calls to `ReadMemory` correctly. But in this update, the else of this `?:` expression is never used because at line 127 you only enter this if block if length is > 0

Reply:
Sorry, my mistake. I have updated it.